deps/build.jl

```diff
@@ -5,94 +5,89 @@
 # print to stderr, since that is where Pkg prints its messages
 eprintln(x...) = println(STDERR, x...)
 
-include("ipython.jl")
-const ipython, ipyvers = find_ipython()
+juliaprofiles = Array(String,0)
 
-if ipyvers < v"1.0.0-dev"
-    error("IPython 1.0 or later is required for IJulia, got $ipyvers instead")
-else
-    eprintln("Found IPython version $ipyvers ... ok.")
-end
+@windows_only begin
+    existing_install_tag_filename = normpath(pwd(),"usr","python342-exists")
+    downloadsdir = normpath(pwd(), "downloads")
+    pythonzipfilename = normpath(pwd(), "downloads", "python-3.4.2.zip")
+    pyinstalldir = normpath(pwd(),"usr","python34")
+    pythonexepath = normpath(pwd(),"usr","python34","python.exe")
+    ijuliaprofiledir = normpath(pwd(), "usr", ".ijulia")
 
-#######################################################################
-# Create Julia profile for IPython and fix the config options.
-
-# create julia profile (no-op if we already have one)
-eprintln("Creating julia profile in IPython...")
-run(`$ipython profile create julia`)
-
-juliaprof = chomp(readall(`$ipython locate profile julia`))
-
-# set c.$s in prof file to val, or nothing if it is already set
-# unless overwrite is true
-function add_config(prof::String, s::String, val, overwrite=false)
-    p = joinpath(juliaprof, prof)
-    r = Regex(string("^[ \\t]*c\\.", replace(s, r"\.", "\\."), "\\s*=.*\$"), "m")
-    if isfile(p)
-        c = readall(p)
-        if ismatch(r, c)
-            m = replace(match(r, c).match, r"\s*$", "")
-            if !overwrite || m[search(m,'c'):end] == "c.$s = $val"
-                eprintln("(Existing $s setting in $prof is untouched.)")
-            else
-                eprintln("Changing $s to $val in $prof...")
-                open(p, "w") do f
-                    print(f, replace(c, r, old -> "# $old"))
-                    print(f, """
-c.$s = $val
-""")
-                end
-            end
-        else
-            eprintln("Adding $s = $val to $prof...")
-            open(p, "a") do f
-                print(f, """
-
-c.$s = $val
-""")
-            end
-        end
+    upgrade_private_python = ispath(existing_install_tag_filename)
+
+    if upgrade_private_python
+        rm(existing_install_tag_filename)
+
+        run(`$pythonexepath -m pip install -U pip`)
+        run(`$pythonexepath -m pip install -U ipython[notebook]==2.3.1`)
     else
-        eprintln("Creating $prof with $s = $val...")
-        open(p, "w") do f
-            print(f, """
-c = get_config()
-c.$s = $val
-""")
+        using BinDeps
+
+        if ispath(downloadsdir)
+            run(`cmd /C RD "$downloadsdir" /S /Q`)
```

Comment:
Ha, I didn't find the ``recursive`` option when I originally looked. Now changed.